test/e2e/prometheusadapter_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"log"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestPrometheusAdapterCARotation(t *testing.T) {
+	// Wait for Prometheus adapter
+	err := wait.Poll(time.Second, 5*time.Minute, func() (bool, error) {
+		_, err := f.KubeClient.Apps().Deployments(f.Ns).Get("prometheus-adapter", metav1.GetOptions{})
+		if err != nil {
+			return false, nil
```

Comment:
Let's surface this error to the user for better debugging. Something along the lines of https://github.com/openshift/cluster-monitoring-operator/pull/240/files.

Reply:
very good idea :+1: